lib/s3.js

```diff
@@ -29,20 +29,21 @@ S3.uploadZip = async function (pathToZip) {
   // Call s3.upload and return it as a promise
   // (NOTE: B.promisify doesn't work for s3.upload, it causes s3/managed_upload.js to throw an error `self.service.constructor.__super__ is not a constructor`)
   logger.debug(`Uploading ${pathToZip} to S3 ${Bucket}`);
-  return await new B(async (resolve, reject) => {
+  let body = await fs.readFile(pathToZip);
+  return await new B((resolve, reject) => {
     const uploader = s3.upload({
       Bucket,
       Key: `${uuid()}.zip`, // Name it with a GUID to avoid conflicts
       ACL: 'public-read',
       Expiration: 1, // Expires after one day
-      Body: await fs.readFile(pathToZip),
+      Body: body,
     }, (err, res) => {
       if (err) {
         reject(new Error(`Could not upload ${pathToZip} to S3. Check that you have set the S3 environment variables correctly: ${err}`));
       } else {
         resolve(res);
-        uploader.on('httpUploadProgress', (progress) => logger.debug(`Uploading ${pathToZip} progress ${Math.round(progress.loaded / progress.total * 100)}%`));
       }
+      uploader.on('httpUploadProgress', (progress) => logger.debug(`Uploading ${pathToZip} progress ${Math.round(progress.loaded / progress.total * 100)}%`));
```

Comment:
this uploader library seems really weird. we're using the return value of `s3.upload` here in the _callback_ of `s3.upload`??

Reply:
I moved it out of the callback. It shouldn't have been in there.